captum/attr/_core/feature_ablation.py

```diff
@@ -601,3 +593,20 @@ def _find_output_mode(
             feature_mask is None
             or all(len(sm.shape) == 0 or sm.shape[0] == 1 for sm in feature_mask)
         )
+
+    def _run_forward(self, *args, **kwargs) -> Tensor:
+        forward_output = _run_forward(*args, **kwargs)
+        if isinstance(forward_output, Tensor):
+            return forward_output
+
+        output_type = type(forward_output)
+        assert output_type is int or output_type is float, (
+            "the return of forward_func must be a tensor, int, or float,"
+            f" received: {forward_output}"
+        )
+
+        # using python type as torch dtype is not a documented behavior
+        # our tests expect int -> torch.int64, float -> torch.float64
+        # but this may actually depend on the machine
+        # ref: https://docs.python.org/3.10/library/stdtypes.html#typesnumeric
+        return torch.tensor(forward_output, dtype=output_type)
```

Comment:
I inherit our original logic that passing python types to torch dtype, like `dtype=float`.  But this is not an officially documented operation. Existing tests assume it must equal to `dtype=torch.float64` https://github.com/pytorch/captum/blob/5f878af6a7/tests/attr/test_feature_ablation.py#L429

But this may be machine dependent https://docs.python.org/3.10/library/stdtypes.html#typesnumeric . 

> Floating point numbers are usually implemented using double in C; information about the precision and internal representation of floating point numbers for the machine on which your program is running is available in [sys.float_info](https://docs.python.org/3.10/library/sys.html#sys.float_info) 

Two other alternatives are:
- explicitly map python type to torch dtype: float -> torch.float64
- do not set dtype, rely on torch's default dtype (float32)

Reply:
Thx for the deep dive @vivekmig ! 
Then I will just add the comment to refer the mapping, also as a caveat.
After all, it is not a documented torch usage. May have breaking changes someday.

Reply:
Makes sense, sounds good!